### PR TITLE
fix: update node annotation used to limit log spam with valid key

### LINKED ISSUE #6300: Annotation used to exclude Nodes from the autoscaler check uses invalid key
**Which component are you using?**: cluster-autoscaler

**What version of the component are you using?**: `HEAD` (feature is unreleased)

Component version: `HEAD`

**What k8s version are you using (`kubectl version`)?**: EKS 1.28

<details><summary><code>kubectl version</code> Output</summary><br><pre>
$ kubectl version

```
Client Version: version.Info{Major:"1", Minor:"27", GitVersion:"v1.27.4", GitCommit:"fa3d7990104d7c1f16943a67f11b154b71f6a132", GitTreeState:"clean", BuildDate:"2023-07-19T12:14:48Z", GoVersion:"go1.20.6", Compiler:"gc", Platform:"darwin/arm64"}
Kustomize Version: v5.0.1
Server Version: version.Info{Major:"1", Minor:"28+", GitVersion:"v1.28.3-eks-4f4795d", GitCommit:"e77944ea667aee90c84c48a74a2a0da7a23e0508", GitTreeState:"clean", BuildDate:"2023-10-20T23:21:39Z", GoVersion:"go1.20.10", Compiler:"gc", Platform:"linux/amd64"}
```

</pre></details>

**What environment is this in?**: AWS

**What did you expect to happen?**:

Applying the annotation to Nodes would reduce the levels of redundant messaging in the controller

**What happened instead?**:

Annotation could not be applied.

**How to reproduce it (as minimally and precisely as possible)**:

```
; k patch node ip-XXX-XXX-XXX-XXX.ec2.internal --patch '{"metadata": {"annotations": {"k8s.io/cluster-autoscaler/enabled": "false"}}}'
The Node "ip-XXX-XXX-XXX-XXX.ec2.internal" is invalid: metadata.annotations: Invalid value: "k8s.io/cluster-autoscaler/enabled": a qualified name must consist of alphanumeric characters, '-', '_' or '.', and must start and end with an alphanumeric character (e.g. 'MyName',  or 'my.name',  or '123-abc', regex used for validation is '([A-Za-z0-9][-A-Za-z0-9_.]*)?[A-Za-z0-9]') with an optional DNS subdomain prefix and '/' (e.g. 'example.com/MyName')
```

**Anything else we need to know?**:

### PR DESCRIPTION
**Pull request by brydoncheyney**
#### What type of PR is this?

/kind bug

#### What this PR does / why we need it:

https://github.com/kubernetes/autoscaler/pull/6265 introduced the ability for users to apply an annotation to cluster Nodes that are not managed by the autoscaler, reducing the amount of "spam" messages logged by the controller.

However, the chosen annotation key `k8s.io/cluster-autoscaler/enabled` is not a valid identifier - https://kubernetes.io/docs/concepts/overview/working-with-objects/annotations/#syntax-and-character-set - which expects only a _single_ DNS prefix.

Attempting to manually add this annotation to an existing node will fail -

```
# nodes "ip-XXX-XXX-XXX-XXX.ec2.internal" was not valid:
# * metadata.annotations: Invalid value: "k8s.io/cluster-autoscaler/enabled": a qualified name must consist of alphanumeric characters, '-', '_' or '.', and must start and end with an alphanumeric character (e.g. 'MyName',  or 'my.name',  or '123-abc', regex used for validation is '([A-Za-z0-9][-A-Za-z0-9_.]*)?[A-Za-z0-9]') with an optional DNS subdomain prefix and '/' (e.g. 'example.com/MyName')
```

Note that the use of the `k8s.io/cluster-autoscaler/enabled` _tag_ is valid within AWS.

This PR uses a valid identifier `k8s.io/cluster-autoscaler-enabled`

#### Which issue(s) this PR fixes: 6300

Fixes #6300

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by drmorr0** [comment]
OH, grumble.

I was thinking the tags got translated into annotations on the node but clearly not.

@gjtempleton can I can a second opinion on this?  I feel less confident in my decision-making abilities now, lol

**Comment by vadasambar** [comment]
DNS regex validation check in kubernetes allows only a single `/`: https://github.com/kubernetes/kubernetes/blob/ec5096fa869b801d6eb1bf019819287ca61edc4d/staging/src/k8s.io/apimachinery/pkg/util/validation/validation.go#L45-L61

**Comment by brydoncheyney** [comment]
> DNS regex validation check in kubernetes allows only a single `/`: [kubernetes/kubernetes@`ec5096f`/staging/src/k8s.io/apimachinery/pkg/util/validation/validation.go#L45-L61](https://github.com/kubernetes/kubernetes/blob/ec5096fa869b801d6eb1bf019819287ca61edc4d/staging/src/k8s.io/apimachinery/pkg/util/validation/validation.go#L45-L61)
> 
> Seems like an easy to trip on thing.

The annotation key/value format is well documented - https://kubernetes.io/docs/concepts/overview/working-with-objects/annotations/#syntax-and-character-set - but I think the confusion here is easy to understand when you consider the _AWS_ tags used by the cluster autoscaler, which do not follow the same conventions and can contain arbitrary characters not support by k8s annotations/labels.

tldr: _easy mistake to make_

**Comment by vadasambar** [comment]
I will take a look if you don't mind @drmorr0 (we can get Guy to do the final review). 

/assign @vadasambar

**Comment by vadasambar** [comment]
LGTM overall. Thank you for the PR. I will do one more pass tomorrow.

**Review by vadasambar — APPROVED** [review]

**Comment by vadasambar** [comment]
@gjtempleton need your approval :pray:

**Comment by gjtempleton** [comment]
Thanks for picking this up, agreed, on first pass my AWS tags brain couldn't see the problem.

/approve

**Pull request merged**
